lib/utils.js

```diff
@@ -0,0 +1,110 @@
+const os = require('os');
+const { Writable } = require('stream');
+const colors = require('chalk');
+
+const isWin = os.platform() === 'win32';
+
+const getType = obj => Object.prototype.toString.call(obj).slice(8, -1);
+
+const NPS_PREFIX = colors.hex('#87E752').bold('NPS>');
+const logger = {
+  info: msg => console.log(`${NPS_PREFIX} ${colors.gray(msg)}`),
+  debug: msg => console.log(`${NPS_PREFIX} ${colors.gray(msg)}`),
+  ok: msg => console.log(`${NPS_PREFIX} ${colors.green(msg)}`),
+  warn: msg => console.log(`${NPS_PREFIX} ${colors.yellow(msg)}`),
+  error: msg => console.log(`${NPS_PREFIX} ${colors.red(msg)}`),
+};
+
+class ShellStreamBuffer extends Writable {
+  constructor(EOI = 'EOI', options) {
```

Comment:
This does not deal well with backpressure which probably isn't a big deal as the stream is a termianl